### PR TITLE
Add the preference query parameter options

### LINKED ISSUE #4034: [DOC] Add documentation for new preferences for search requests
**What do you want to do?**
 
- [ ] Request a change to existing documentation
- [x] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 As a part of https://github.com/opensearch-project/OpenSearch/pull/7375, we added a new options for preferences to search requests - `_primary`, `_replica`, `_primary_first` and `_replica_first`. 

None of the preferences are listed out on `https://opensearch.org/docs/2.7/api-reference/search/` (Request parameters -> Preferences) but it would be useful for the clients and users to have a list of available preferences. 

```
_primary - Search will be performed only on the primary shards.
_replica - Search will be performed only on the replica shards. 
_primary_first - Search will be performed preferably on the primary shards, but can failover to other available shards. 
_replica_first - Search will be performed preferably on the replica shards, but can failover to other available shards including primary. 
```

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.
- https://github.com/opensearch-project/OpenSearch/pull/7375
- https://github.com/opensearch-project/OpenSearch/issues/6046

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Fixes #4034 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by kotwanikunal — APPROVED** [review]

File: _api-reference/search.md [hunk 1]
```diff
@@ -60,7 +60,7 @@ ignore_unavailable | Boolean | Specifies whether to include missing or closed in
 lenient | Boolean | Specifies whether OpenSearch should accept requests if queries have format errors (for example, querying a text field for an integer). Default is false.
 max_concurrent_shard_requests | Integer | How many concurrent shard requests this request should execute on each node. Default is 5.
 pre_filter_shard_size | Integer | A prefilter size threshold that triggers a prefilter operation if the request exceeds the threshold. Default is 128 shards.
-preference | String | Specifies which shard or node OpenSearch should perform the count operation on.
+preference | String | Specifies on which shard or node OpenSearch should perform the search. For valid values, see [The `preference` query parameter](#the-preference-query-parameter).
```
Comment (cwillum) [inline _api-reference/search.md:63]:
```suggestion
preference | String | Specifies the shards or nodes on which OpenSearch should perform the search. For valid values, see [The `preference` query parameter](#the-preference-query-parameter). 
```

File: _api-reference/search.md [hunk 2]
```diff
@@ -86,6 +86,20 @@ track_total_hits | Boolean or Integer | Whether to return how many documents mat
 typed_keys | Boolean | Whether returned aggregations and suggested terms should include their types in the response. Default is true.
 version | Boolean | Whether to include the document version as a match.
 
+### The `preference` query parameter
+
+The `preference` query parameter specifies on which shard or node OpenSearch should perform the search. Valid values are:
```
Comment (cwillum) [inline _api-reference/search.md:91]:
```suggestion
The `preference` query parameter specifies the shards or nodes on which OpenSearch should perform the search. Valid values are:
```

**Review by cwillum — APPROVED** [review]
Looks good.

File: _api-reference/search.md [hunk 3]
```diff
@@ -86,6 +86,20 @@ track_total_hits | Boolean or Integer | Whether to return how many documents mat
...
 
+### The `preference` query parameter
+
+The `preference` query parameter specifies the shards or nodes on which OpenSearch should perform the search. Valid values are:
+
+- `_primary`: Perform the search only on primary shards.
+- `_replica`: Perform the search only on replica shards.
+- `_primary_first`: Perform the search on primary shards, but fail over to other available shards if primary shards are not available.
+- `_replica_first`: Perform the search on replica shards, but fail over to other available shards if replica shards are not available.
+- `_local`: If possible, perform search on the local node's shards.
```
Comment (hdhalter) [inline _api-reference/search.md:97]:
```suggestion
- `_local`: If possible, perform the search on the local node's shards.
```

File: _api-reference/search.md [hunk 4]
```diff
@@ -86,6 +86,20 @@ track_total_hits | Boolean or Integer | Whether to return how many documents mat
...
+- `_prefer_nodes:<node-id-1>,<node-id-2>`: If possible, perform search on the specified nodes. Use a comma-separated list to specify multiple nodes.
```
Comment (hdhalter) [inline _api-reference/search.md:98]:
```suggestion
- `_prefer_nodes:<node-id-1>,<node-id-2>`: If possible, perform the search on the specified nodes. Use a comma-separated list to specify multiple nodes.
```

**Review by hdhalter — COMMENTED** [review]
Just a few tweaks for consistency.

File: _api-reference/search.md [hunk 5]
```diff
@@ -86,6 +86,20 @@ track_total_hits | Boolean or Integer | Whether to return how many documents mat
 typed_keys | Boolean | Whether returned aggregations and suggested terms should include their types in the response. Default is true.
 version | Boolean | Whether to include the document version as a match.
 
+### The `preference` query parameter
+
+The `preference` query parameter specifies the shards or nodes on which OpenSearch should perform the search. Valid values are:
+
+- `_primary`: Perform the search only on primary shards.
+- `_replica`: Perform the search only on replica shards.
+- `_primary_first`: Perform the search on primary shards, but fail over to other available shards if primary shards are not available.
```
Comment (natebower) [inline _api-reference/search.md:95]:
```suggestion
- `_primary_first`: Perform the search on primary shards but fail over to other available shards if primary shards are not available.
```

File: _api-reference/search.md [hunk 6]
```diff
@@ -86,6 +86,20 @@ track_total_hits | Boolean or Integer | Whether to return how many documents mat
...
+- `_replica_first`: Perform the search on replica shards, but fail over to other available shards if replica shards are not available.
```
Comment (natebower) [inline _api-reference/search.md:96]:
```suggestion
- `_replica_first`: Perform the search on replica shards but fail over to other available shards if replica shards are not available.
```

File: _api-reference/search.md [hunk 7]
```diff
@@ -86,6 +86,20 @@ track_total_hits | Boolean or Integer | Whether to return how many documents mat
...
+- `_local`: If possible, perform the search on the local node's shards.
+- `_prefer_nodes:<node-id-1>,<node-id-2>`: If possible, perform the search on the specified nodes. Use a comma-separated list to specify multiple nodes.
+- `_shards:<shard-id-1>,<shard-id-2>`: Perform the search only on the specified shards. Use a comma-separated list to specify multiple shards. When combined with other preferences, the `_shards` preference must be listed first. For example, `_shards:1,2|_replica`.
+- `_only_nodes:<node-id-1>,<node-id-2>`: Perform the search only on the specified nodes. Use a comma-separated list to specify multiple nodes.
+- `<string>`: Specifies a custom string to use for search. The string cannot start with an underscore character (`_`). Searches with the same custom string are routed to the same shards.
```
Comment (natebower) [inline _api-reference/search.md:101]:
```suggestion
- `<string>`: Specifies a custom string to use for the search. The string cannot start with an underscore character (`_`). Searches with the same custom string are routed to the same shards.
```

File: _api-reference/search.md [hunk 8]
```diff
@@ -86,6 +86,20 @@ track_total_hits | Boolean or Integer | Whether to return how many documents mat
 typed_keys | Boolean | Whether returned aggregations and suggested terms should include their types in the response. Default is true.
 version | Boolean | Whether to include the document version as a match.
 
+### The `preference` query parameter
+
+The `preference` query parameter specifies the shards or nodes on which OpenSearch should perform the search. Valid values are:
```
Comment (natebower) [inline _api-reference/search.md:91]:
```suggestion
The `preference` query parameter specifies the shards or nodes on which OpenSearch should perform the search. The following are valid values:
```

Comment (natebower) [inline _api-reference/search.md:91]: Where possible, the text preceding the colon should be a complete clause.

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Just a few minor changes. Thanks!

**Pull request merged**